### ISSUE TITLE
OAuth realms roles and client roles do not work together

### ISSUE THREAD

**Issue by andreasfranz** [jhipster/generator-jhipster#11361]
##### **Overview of the issue**

In our setup we use Keycloak as SSO server for OAuth. If you use realm roles together with client roles the JHipster application is unable to obtain the client roles

##### **Motivation for or Use Case**

A combination of realm and client roles should be possible since the are just roles with different scope.

##### **Reproduce the error**

Configure both realm and client roles for a user. The client roles will not be available to the application.

##### **JHipster Version(s)**
6.6.0

##### **JHipster Version(s)**

```
#######@0.0.1-SNAPSHOT ######
└── generator-jhipster@6.6.0 

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "#####",
      "nativeLanguage": "de",
      "microservicePath": "../#####/"
    },
    "jhipsterVersion": "6.6.0",
    "applicationType": "gateway",
    "baseName": "#####",
    "packageName": "#####",
    "packageFolder": "#####",
    "serverPort": "8082",
    "authenticationType": "oauth2",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "embeddableLaunchScript": false,
    "useSass": true,
    "clientPackageManager": "yarn",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en", "de"],
    "blueprints": [],
    "creationTimestamp": 1579530224774
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

openjdk version "1.8.0_242"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_242-b08)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.242-b08, mixed mode)

git version 2.25.0

node: v13.1.0

npm: 6.12.1

yeoman: 2.0.0

yarn: 1.22.0

Docker version 19.03.5, build 633a0ea

docker-compose version 1.25.4, build 8d51620a

INFO! Congratulations, JHipster execution is complete!

##### **Browsers and Operating System**

macOS Catalina 10.15.3
Chrome 80.0.3987.116

-   [ x] Checking this box is mandatory (this is just to show you read everything)

**Comment by mraible** [comment]
This seems like a Keycloak specific issue. We like to keep our OAuth support generic do it works with all providers.

**Comment by andreasfranz** [comment]
@mraible in which way does this break any generic OAuth provider?

**Comment by mraible** [comment]
@andreasfranz After looking at the code, I don't think it does. It's just changing the logic from allowing a `roles` OR a `groups` claim to looking for both and combining them. I'm OK with it, but you should add "Fixes #11361" to the description of https://github.com/jhipster/generator-jhipster/pull/11362 so your PR is related to this issue.
